### PR TITLE
Remove Mixer from the Istio codebase

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR removes all Mixer-related code from the repository, including references to Mixer config and APIs (in validation and other components).

Mixer was deprecated several releases ago. Release 1.7 is the last release in which Mixer is still shipped as a component of Istio. As per the [Mixer Removal Plan](https://docs.google.com/document/d/1abYN18B8JyV64EFHtNm0uSIM4DNRX8JddJiwacNBoHg/edit#heading=h.qex63c29z2to), this PR is removing the code from identified sections of the codebase.

Mixer has been replaced by proxy-based extensions for Istio. An extensions eco-system will replace the extensibility functionality offered by Mixer, powered by Wasm. Standard Istio extensions, including the Prometheus plugin, have already been converted to the new model (this work is commonly referred to as "telemetry v2").

A [shim for existing Mixer gRPC adapters](https://github.com/istio/istio/issues/23580) in under [active development](https://github.com/istio/istio/pull/25624), and will be delivered in the 1.7.x timeframe. This will enable those that still wish to use Mixer to continue to do so (albeit a Mixer frozen in time). However, users are _strongly_ encouraged to transition to the new model for Istio extension.

The companion to this PR for the mixer client code is https://github.com/istio/proxy/pull/2936. Another PR, for the istio/api repo, will follow to completely remove the Mixer APIs.

WARNING: thrift-based rate limiting relied on the deprecated (and now removed) mixer client `QuotaSpec` and `QuotaSpecBinding` APIs. **This PR disables that support within Istio.** A new API surface for that functionality will need to be developed and added back to support thrift-based rate limiting. Tracking bug: https://github.com/istio/istio/issues/25725.

---

Mixer (and its prototype predecessor) was part of the original proof-of-concept work that, in part, gave birth to the Istio project. It served several of its original purposes fairly well. Mixer enabled (relatively) easy extension of Istio for policy and telemetry, at a time before many of the now overlapping Envoy APIs were created and/or matured. Mixer also enabled simple operator-driven customization of policy enforcement and telemetry generation by providing fine-grained, attribute-based, configuration APIs. It fell short of the original goals in other areas however -- primarily related to scaling and performance.

Since Mixer's creation, several Envoy APIs (such as [ext-authz](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html)) and the entirely new [proxy-wasm](https://github.com/proxy-wasm/spec) project have emerged to fill similar functions. These new technologies enable Istio extension in a more performant and scalable manner, while still providing a (relatively) simple developer story.

And so it is time to say goodbye to Mixer.

A big thank you to all the contributors that have helped build and maintain Mixer throughout the years, including: @geeknoid, @ZackButcher, @guptasu, @ozevren, @qiwzhang, @mandarjog, @kyessenov, @bianpengyuan, @gargnupur, and many others.

Addresses: #25333.

[ X ] Networking
[ X ] Policies and Telemetry
[ X ] Test and Release

### CODE REVIEW

**Comment by kyessenov** [comment]
Are you also doing it in istio/proxy?

File: bin/check_samples.sh [hunk 1]
```diff
@@ -28,9 +28,6 @@ go build -o $ISTIOCTL ./istioctl/cmd/istioctl
 for f in samples/**/*.yaml; do
   echo "Validating $f..."
   $ISTIOCTL validate -x \
-    -f mixer/testdata/config/attributes.yaml \
-    -f mixer/template/metric/template.yaml \
-    -f mixer/test/prometheus/prometheus-nosession.yaml \
     -f samples/httpbin/policy/keyval-template.yaml \
```
Comment (kyessenov) [inline bin/check_samples.sh:31]: Remove this, too, and the associated sample.

**Comment by douglas-reid** [comment]
> Are you also doing it in istio/proxy?

Yes.

File: bin/check_samples.sh (hunk 1 above)
Comment (douglas-reid) [reply bin/check_samples.sh]: good catch, thanks.

**Review by howardjohn — COMMENTED** [review]
I've reviewed the changes alphabetically before `mixer` and lgtm to me so far. Will review the rest tomorrow

**Comment by douglas-reid** [comment]
> I've reviewed the changes alphabetically before `mixer` and lgtm to me so far. Will review the rest tomorrow

that's fairly impressive. fwiw, you can skip the `mixer/` section, as that is pure deletion.

**Comment by mandarjog** [comment]
@douglas-reid  nice branch name.

This is one giant effort.

So far I found the following missing

```
tests/codecov/main_test.go
tests/testdata/config/none.yaml
operator/cmd/mesh/shared.go
```

**Comment by mandarjog** [comment]
```
mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
$ vi operator/pkg/translate/testdata/icp-iop/input/default.yaml
mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
$ grep -i -r mixer * -l --include='*.yaml' | wc -l
87
mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
$ grep -i -r mixer * -l --include='*.json' | wc -l
4
mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
$ grep -i -r mixer * -l --include='*.go' | wc -l
8
mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
$ grep -i -r mixer * -l --include='*.go'
operator/cmd/mesh/upgrade.go
operator/cmd/mesh/shared.go
operator/pkg/apis/istio/v1alpha1/values_types.pb.go
operator/pkg/apis/istio/v1alpha1/validation/validation.go
operator/pkg/apis/istio/v1alpha1/validation/validation_test.go
operator/pkg/translate/translate.go
operator/pkg/translate/translate_value.go
operator/pkg/translate/translate_value_test.go
```

**Comment by howardjohn** [comment]
The operator stuff is done in https://github.com/istio/istio/pull/25363
since it's tangled with the addons

**Comment by douglas-reid** [comment]
> ```
> mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
> $ vi operator/pkg/translate/testdata/icp-iop/input/default.yaml
> mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
> $ grep -i -r mixer * -l --include='*.yaml' | wc -l
> 87
> mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
> $ grep -i -r mixer * -l --include='*.json' | wc -l
> 4
> mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
> $ grep -i -r mixer * -l --include='*.go' | wc -l
> 8
> mjog@mjog-macbookpro2 ~/tmp/mixer-removal/istio (goodnight-sweet-prince)
> $ grep -i -r mixer * -l --include='*.go'
> operator/cmd/mesh/upgrade.go
> operator/cmd/mesh/shared.go
> operator/pkg/apis/istio/v1alpha1/values_types.pb.go
> operator/pkg/apis/istio/v1alpha1/validation/validation.go
> operator/pkg/apis/istio/v1alpha1/validation/validation_test.go
> operator/pkg/translate/translate.go
> operator/pkg/translate/translate_value.go
> operator/pkg/translate/translate_value_test.go
> ```

as @howardjohn relates, this PR doesn't touch `manifests/` or `operator/` (awaiting the other PR).

**Comment by douglas-reid** [comment]
/retest

and then go to bed.

**Comment by howardjohn** [comment]
Adding release-notes-note to bypass a bug in the test so we get a green checkmark to avoid scaring reviewers

**Review by howardjohn — APPROVED** [review]
I reviewed this to ensure that all changes made are what we want,  and we didn't delete anything we still need. To that end, this looks great.

I didn't focus much on ensuring we deleted every single reference to mixer, as I think we can easily followup with more removals.

LGTM!

RIP mixer

**Review by mandarjog — APPROVED** [review]
Lgtm

**Comment by nrjpoddar** [comment]
Kudos @douglas-reid on the amazing name (and the PR of course) :P

**Review by nmittler — APPROVED** [review]

**Review by JimmyCYJ — APPROVED** [review]

**Review by ericvn — APPROVED** [review]
I have done some initial looking at how much impact this will have on the istio.io repo when it branches to 1.7 and we move it to refer back to `master` as we will possibly see a number of missing links. 

The majority of the pages affected already have _deprecated_ on the page so removal should be OK. Taking that one step further (pages who refer to the removed pages) looks pretty limited as well. Most of the items will be features (listed as beta or stable) being removed from the table.

**Review by myidpt — APPROVED** [review]

**Pull request merged**